stdlib/domain.mli

```diff
@@ -105,6 +105,14 @@ module DLS : sig
         domain will be computed on-demand by the new domain: the first
         [get] call will call the initializer [f] and store that value.
 
+        {b Warning.} [f] may be called several times if several
+        threads on the same domain try to access the key
+        concurrently. Only the 'first' value computed will be used,
```

Comment:
Isn't that overly specific? What about signal handlers, finalizers, etc? What if `f` itself is using DLS?

Reply:
New wording, what do you think?

        {b Warning.} [f] may be called several times if another call
        to [get] occurs during initialization on the same domain. Only
        the 'first' value computed will be used, the other now-useless
        values will be discarded. Your initialization function should
        support this situation, or contain logic to detect this case
        and fail.